harness/tests/test_ipc.py

```diff
@@ -158,3 +163,109 @@ def test_zmq_server_client() -> None:
     server.send(server_object)
     client_object = client.receive()
     assert server_object == client_object
+
+
+@pytest.mark.parametrize("cross_size", [1, 4])  # type: ignore
+@pytest.mark.parametrize("local_size", [1, 4])  # type: ignore
+@pytest.mark.parametrize("force_tcp", [False, True])  # type: ignore
+def test_distributed_context(cross_size: int, local_size: int, force_tcp: bool) -> None:
+    size = cross_size * local_size
+    # Generate one rendezvous_info per node.
+    rendezvous_info = [
+        det.RendezvousInfo(
+            addrs=["localhost:12345"] * cross_size,
+            rank=i,
+        )
+        for i in range(cross_size)
+    ]
+
+    def do_parallel(fn: Callable) -> List:
+        """
+        Run the same function on one-thread-per-rank, assert there were no exceptions, and return
+        the results from each rank.
+        """
+        results = [None] * size  # type: List
+        errors = [None] * size  # type: List
+        threads = []
+
+        for cross_rank, local_rank in itertools.product(range(cross_size), range(local_size)):
+            rank = cross_rank * local_size + local_rank
+
+            def _fn(rank: int, cross_rank: int, local_rank: int) -> None:
```

Comment:
```suggestion
            def _run_fn(rank: int, cross_rank: int, local_rank: int) -> None:
```

Reply:
I think since `_fn` wraps `fn`, it is a clearer name than `_run_fn`

Reply:
It's weird a `fn` wraps `_fn`. Shouldn't `fn` wrap `_fn`?

Reply:
no... `fn` is a parameter, and those shouldn't have leading underscore names.  `_fn` wraps the parameter named `fn`.

Reply:
I am fine with this name since it's only one place.